app/assets/javascripts/beak/widgets/ractives/info.coffee

```diff
@@ -40,7 +40,7 @@ window.RactiveInfoTabWidget = Ractive.extend({
     # html_sanitize is provided by Google Caja - see https://code.google.com/p/google-caja/wiki/JsHtmlSanitizer
     # RG 8/18/15
     window.html_sanitize(
-      exports.toHTML(md),
+      window.markdown.toHTML(md),
```

Comment:
Directly importing `markdown-js` would definitely be nicer, but I found a few issues on the way:

1. `markdown-js` doesn't support the standard `esm` module format used in browsers. It works either in the browser using a global variable or using the Node.js module format `commonjs`. This *can* be solved by using [@rollup/plugin-commonjs](https://www.npmjs.com/package/@rollup/plugin-commonjs).

2. The bigger issue is, that `markdown-js` uses the Node.js `util` api, when used in the `commonjs` format. There would even be an option to polyfill `util` using [rollup-plugin-node-polyfill](https://www.npmjs.com/package/rollup-plugin-node-polyfills), but then the entire `util` api would be included in the bundle (markdown-js doesn't use named imports when requiring `util` and so the unused parts of `util` cannot be removed by Rollup's tree-shaking).
    
Using multiple plugins and adding unused code from `util` to the bundle to avoid 1 global variable is probably not a good idea, so I left it as it was for now.
    
But we could maybe use [rollup-plugin-external-globals](https://www.npmjs.com/package/rollup-plugin-external-globals), which would allow us to `import` global libraries (such as markdown-js, google-caja, JQuery, Racive.js etc.). The advantage of this would be it makes dependencies of a script file more obvious and refactors would be a little bit safer. Do you think using a plugin for this would be worth it?

Reply:
I'm fine leaving this one library as an old-style import.  Rather than worrying about shimming in this library if we did decide it was important to have it import as a module, I'd rather seek out a more up-to-date and maintained JS Markdown package to use (I'm sure there are loads of them).

Reply:
Yikes!  I definitely don't like the idea of bringing in `util` for the sake of importing `markdown-js`.

I'm fine with leaving things as they are, and maybe updating the library at some point.  That said, if it's not difficult to set up, I think that I honestly find rollup-plugin-external-globals to be most appealing, since we'd then have a clear declaration of where a bunch of global variables/function names are coming from.  That sort of thing often trips me up and leaves me wondering things like, "Where did I define this `markdown` global all those years ago...? :thinking: "

And that's not a hypothetical example, by the way; it's what I did as I was writing this code review, and it took me a little bit to figure out that it was coming from a library. :laughing:  Having that explicitly stated somewhere would have sped that process up a bit for me.

Reply:
Alright, I'll leave it global for now. Adding [rollup-plugin-external-globals](https://www.npmjs.com/package/rollup-plugin-external-globals) should be pretty straightforward, but if we want to `import` as many of the used libraries as possible, it would take some work finding and replacing all the references to global variables with imports.

---

Sidenote:

I took a quick look at the dependencies in `package.json` (+ google-caja and markdown-js from webjars) and categorized them by what they would need to be `import`ed (judging just from their docs, I didn't try anything out):

These are libraries, that probably need to remain global, but could be `import`ed with [rollup-plugin-external-globals](https://www.npmjs.com/package/rollup-plugin-external-globals):
- markdown-js (for the reasons explained above)
- google-caja
- codemirror (needs external CSS)
- jquery (has to be a global, because the chosen-js plugin needs it)
- nettango
- synchrodecoder (available just through `window`)
- jscolor-picker (@LaCuneta said it is no longer needed and will be removed)

These libraries provide an esm bundle (yay!) and could be imported directly from node_modules using [@rollup/plugin-node-resolve](https://www.npmjs.com/package/@rollup/plugin-node-resolve):
- ractive

And these libraries could probably be imported directly from mode_modules but would require @rollup/plugin-commonjs(https://www.npmjs.com/package/@rollup/plugin-commonjs) to transpile them from `commonjs` to the `esm` format:
- file-saver
- highcharts
- localforage (I didn't find any references to it, is it still used?)
- mousetrap